### PR TITLE
TIP-221: remove AttributeManager, ProductManager and FamilyManager

### PR DESCRIPTION
**Pull request by jjanvier**
Remove AttributeManager, ProductManager and FamilyManager to next ease Domain layer extraction.

```
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Changelog updated                 | Y
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -
```

### CODE REVIEW

**Comment by nidup** [comment]
:fire: :heart: :fire: :heart:

File: src/Pim/Bundle/CatalogBundle/Manager/AttributeOptionsSorter.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * Sort attribute options and save them.
+ *
+ * @author    Filips Alpe <filips@akeneo.com>
+ * @copyright 2013 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class AttributeOptionsSorter
+{
+    /** @var BulkSaverInterface */
+    private $optionSaver;
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/Manager/AttributeOptionsSorter.php]: private?

File: src/Pim/Bundle/EnrichBundle/Controller/AttributeController.php [hunk 2]
```diff
@@ -90,6 +88,12 @@ class AttributeController
     /** @var GroupRepositoryInterface */
     protected $groupRepository;
 
+    /** @var AttributeTypeRegistry */
+    protected $registry;
+
+    /** @var AttributeFactory */
+    protected $factory;
```
Comment (nidup) [inline src/Pim/Bundle/EnrichBundle/Controller/AttributeController.php]: you'll have a conflict with the @juliensnz PR related to simple factory

File: src/Pim/Bundle/EnrichBundle/Resources/config/datagrid/association_product.yml [hunk 3]
```diff
@@ -92,7 +92,8 @@ datagrid:
                     options:
                         field_options:
                             multiple: true
-                            choices:  '@pim_catalog.manager.family->getChoices'
+                            attr:
+                                empty_choice: true
```
Comment (nidup) [inline src/Pim/Bundle/EnrichBundle/Resources/config/datagrid/association_product.yml]: :dancer:

File: src/Pim/Bundle/EnrichBundle/Resources/config/datagrid/product_group.yml [hunk 4]
```diff
@@ -95,7 +95,8 @@ datagrid:
                     options:
                         field_options:
                             multiple: true
-                            choices:  '@pim_catalog.manager.family->getChoices'
+                            attr:
+                                empty_choice: true
```
Comment (nidup) [inline src/Pim/Bundle/EnrichBundle/Resources/config/datagrid/product_group.yml]: :dancer:

File: src/Pim/Bundle/EnrichBundle/Resources/config/datagrid/product_variant_group.yml [hunk 5]
```diff
@@ -96,7 +96,8 @@ datagrid:
                     options:
                         field_options:
                             multiple: true
-                            choices:  '@pim_catalog.manager.family->getChoices'
+                            attr:
+                                empty_choice: true
```
Comment (nidup) [inline src/Pim/Bundle/EnrichBundle/Resources/config/datagrid/product_variant_group.yml]: :dancer:

**Comment by nidup** [comment]
baby comment on a private, double check your BC break list then GTM

**Comment by jjanvier** [comment]
BC breaks are OK :)

File: src/Pim/Bundle/EnrichBundle/Controller/AttributeController.php (hunk 2 above)
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Controller/AttributeController.php]: it's merged now. You should use it :)

**Comment by juliensnz** [comment]
GTM. Much love, such burn, so kiss

**Comment by jjanvier** [comment]
one behat failing on EE odm
same behat as master, so merged

**Pull request merged**